packages/editor/package.json

```diff
@@ -15,6 +15,7 @@
   ],
   "dependencies": {
     "@axonivy/editor-icons": "11.3.0-next",
+    "@axonivy/inscription-core": "11.3.0-next",
```

Comment:
why is this dependency needed, if the only thing that is now needed from this package is the Deferred util, and it is currently only needed in the editor? Why not move this Deferred util directly to the editor package?
Why is this Deferret util even needed? I think I don't fully understand the purpose of it 🙈

Reply:
We can absolutely move it into the `editor` package as well if you prefer. It really is just such a fundamental utility class that we wanted to put it somewhere more general but unless somebody else uses it we can definitely move it into the editor package.

The idea behind the `Deferred` is a common pattern in many frameworks but it is basically just convenience "manager" for a single promise. When you create a promise you get the `resolve` and `reject` functions that you can call to work with the promise but those functions are not exposed and cannot be called from the outside so you would need to implement all your logic within the promise constructor. So just exposing those functions already gives you a lot more flexibility with the code as you decouple the promise creation from its resolution. You can put additional meta information in the Deferred as well such as the current state of the promise.

So if you look at our use case below, you'll see we immediately create the Deferred object so anybody can already call `getInstance()` to get the future promise without any async code already working on resolving that promise. That only happens after `configureMonaco` is called and we actually resolve the promise. That is not to say you cannot do this differently.

Reply:
I think I would move it into the editor package for now 👍

Reply:
Thanks for your explanation 👍

Reply:
I moved it into the editor package in a new commit.